### DISCUSSION TITLE
How to access response object of a referenced request?

### DISCUSSION THREAD

**Discussion by alekdavis** [AnWeber/vscode-httpyac#331] [Q&A] [answered]
If a test references another test (via `@ref` or `@forceRef`), is there a way to access the HTTP response object of the referenced request? 

Say, I have `createTest` and `updateTest` and I want `updateTest` to be executed only if `createTest` is successful. The easiest way to check would be by making sure `createTest` returns a successful HTTP status (200, 201, or whatever in the 200 range). I can test status in the post-`createTest` assertion, but how do I access this from the pre-`updateTest` script? I can see the response data attached to the `createTest` request in the httpYac tab, but I printed the members of the `createTest` variable and I do not see anything resembling the HTTP response object. I suspect this can be done by saving the value in some variable in the post-request script, but I am wondering if there is an easier way. 

It would be really cool if I could do something like this:

```js
{{
  exports.$cancel = (createTest.__response.status != 200);
}}
```

Or maybe :

```js
{{
  exports.$cancel = (__responses["createTest"].status != 200);
}}
```

**Comment by AnWeber** [comment, marked as answer]
I already create a variable for that. after executing a named httpregion, httpyac creates two variables. Variable `{{name}}` (= response body) and `{{nameResponse}}` (= response'). For `@ref foo` you can use `fooResponse`.
But I actually already set skipped when the @ref has a test error. In other words, your cancel should not actually be necessary

**Comment by alekdavis** [comment]
I just have two words: wow. And wow! That is so awesome. Thank you.

**Comment by alekdavis** [comment]
@AnWeber RE: _"But I actually already set skipped when the @ref has a test error. In other words, your cancel should not actually be necessary."_ 

Unfortunately, it does not seem to be the case. I was just smoke testing a production deployment and in my CRUD test sequence, I have a PATCH call, followed by GET to make sure PATCH actually made an update. So, in the GET call I reference the PATCH test and even though PATCH failed (it expected response HTTP code 200, but 403 was returned), the GET test was still executed. 

Here is abbreviated sequence (with some info redacted):

```javascript
### UPDATE xxx (multiple changes)
# @name updateXxx
# @ref createXxx
PATCH /xxx
Authorization: oauth2 client_credentials clientCredentials
Content-Type: application/json
{
...
}

?? status == 200

### GET xxx after UPDATE (multiple changes)
# @name getXxxAfterUpdate
# @ref updateXxx
GET /xxx/{{createXxx.id}}
Authorization: oauth2 client_credentials clientCredentials

?? status == 200
?? ...
```
My understanding is that any failed assertion of the referenced test should cancel test execution, but it does not seem to be the case. Or did I miss something?

**Comment by alekdavis** [comment]
This is not an issue. The behavior of the conditional execution based on referenced test result changed and it makes more sense now. This conversation is no longer relevant.
